packages/nimble-components/src/table/specs/table-spec.md

```diff
@@ -0,0 +1,242 @@
+
+-   Data API :
+    -   Define what the interface is for setting/getting data on the table, taking into account any framework-specific binding concerns
+    -   Illustrate how the data is hooked up to the TanStack API
+    -   How is data hierarchy provided and represented in the DOM
+-   Column Definitions
+    -   Define the interface we will provide for the column providers/components (i.e., width, sorting, allowSort, allowGrouping, etc...)
+    -   Define the base implementation for the column providers that other column providers can extend.
+    -   List the set of column providers that Nimble will provide and provide their respective APIs where unique (e.g., formatter for DateTime column)
+    -   Define the table-level APIs for setting the columns
+        -   Slot API on table?
+        -   Alternative/accompanying programmatic API?
+-   Headers
+    -   Define the anatomy of headers in the table DOM
+        -   Require specific component type?
+        -   What is the component to use for interaction? Outline Button? Ghost button?
+        -   What and where are the interactive mechanisms/indicators? Sort arrow, etc..
+-   Row Selection
+    -   Define the anatomy of row selection in the table DOM
+        -   Indeterminate checkbox at the far left of each row?
+        -   Selected row CSS
+    -   Define events raised when row selection changes/occurs
+    -   Define table-level row-selection API
+-   Grouping
+    -   Define interactive mechanism, if any, to provide grouping
+    -   Define table-level API for setting grouping
+    -   Define events raised when grouping changes
+-   Sorting
+    -   Define the table-level API for setting sorting
+    -   Define events raised when sorting changes
+-   Action Menu
+    -   Define how the action menu gets associated with a particular column
+    -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
+
+_Properties_
+
+Placeholder
+
+_Events_
+
+Placeholder
+
+### Anatomy
+
+_Slots_
+
+-   `slottedColumns` (Placeholder for column slots)
+-   `actionMenu` (Placeholder for action menu)
+
+### Angular integration
+
+Angular support should be accomplished through the typical directive patterns.
+
+### Blazor integration
+
+Blazor support should be accomplished through the typical integration patterns. It should be noted that if the Nimble `Table` component is generic, then the Blazor component will mirror that generic API.
+
+---
+
+## Implementation
+
+The `nimble-table` will require various sub-components to exist in order to properly/efficiently render the data. The web components and their respective responsibilities are as follows:
+
+`<nimble-table>`
+
+-   Creates the row elements that will be used to render each row of data
+-   Provides any necessary elements for proper virtualization
+-   Manages keyboard navigation between rows
+-   Supports slotting of column elements
+
```

Comment:
Should there be something here about `<nimble-table-column>`?

Reply:
Not yet. I am expecting the Column Customization HLD to cover that, and this section can be updated as needed then.